specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json

```diff
@@ -0,0 +1,376 @@
+          {
+            "name": "QueryPackTags",
+            "description": "Updated tag information to set into the QueryPack instance.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/TagsResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updating the Log Analytics QueryPack's tags was successful. QueryPack tags are updated and returned with the rest of the QueryPack's object properties.",
+            "schema": {
+              "$ref": "#/definitions/LogAnalyticsQueryPack"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "QueryPackUpdateTagsOnly": {
+            "$ref": "./examples/QueryPacksUpdateTagsOnly.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "QueryPacksResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Azure resource name",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location",
+          "x-ms-mutability": ["create", "read"]
+        },
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "required": ["location"],
+      "x-ms-azure-resource": true,
+      "description": "An azure resource object"
+    },
+    "TagsResource": {
+      "properties": {
+        "tags": {
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags"
+        }
+      },
+      "description": "A container holding only the Tags for a resource, allowing the user to update the tags on a QueryPack instance."
+    },
+    "LogAnalyticsQueryPack": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties that define a Log Analytics QueryPack resource.",
+          "$ref": "#/definitions/LogAnalyticsQueryPackProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/QueryPacksResource"
+        }
+      ],
+      "required": ["properties"],
+      "description": "An Log Analytics QueryPack definition."
+    },
+    "LogAnalyticsQueryPackProperties": {
+      "description": "Properties that define a Log Analytics QueryPack resource.",
+      "properties": {
+        "queryPackId": {
+          "type": "string",
+          "description": "The unique ID of your application. This field cannot be changed.",
+          "readOnly": true
+        },
+        "name": {
```

Comment:
Is this different from the top-level name property? If so, a name like displayName might be more appropriate. If it is the same as the top-level name property, it's redundant and should be removed. Does it really need to be unique?

Reply:
thanks, your'e right, removed. resource name is unique within resource group. uniqueness is achieved by resource id.